src/Http/Http.Extensions/src/HttpRequestJsonExtensions.cs

```diff
@@ -68,15 +72,25 @@ public static class HttpRequestJsonExtensions
         options ??= ResolveSerializerOptions(request.HttpContext);
 
         var encoding = GetEncodingFromCharset(charset);
-        var (inputStream, usesTranscodingStream) = GetInputStream(request.HttpContext, encoding);
+        Stream? inputStream = null;
 
         try
         {
+            if (encoding == null || encoding.CodePage == Encoding.UTF8.CodePage)
+            {
+                if (_useStreamJsonOverload)
+                {
+                    return await JsonSerializer.DeserializeAsync<TValue>(request.Body, options, cancellationToken);
+                }
+                return await JsonSerializer.DeserializeAsync<TValue>(request.BodyReader, options, cancellationToken);
+            }
+
+            inputStream = Encoding.CreateTranscodingStream(request.Body, encoding, Encoding.UTF8, leaveOpen: true);
             return await JsonSerializer.DeserializeAsync<TValue>(inputStream, options, cancellationToken);
```

Comment:
There are 3 `awaits` in the method body. In order to reduce compiled code size this could be written as
```suggestion
            Task<TResult> task;
            
            if (encoding == null || encoding.CodePage == Encoding.UTF8.CodePage)
            {
                if (_useStreamJsonOverload)
                {
                    task = JsonSerializer.DeserializeAsync<TValue>(request.Body, options, cancellationToken);
                }
                task = JsonSerializer.DeserializeAsync<TValue>(request.BodyReader, options, cancellationToken);
            }
            else
            {
                inputStream = Encoding.CreateTranscodingStream(request.Body, encoding, Encoding.UTF8, leaveOpen: true);
                task = JsonSerializer.DeserializeAsync<TValue>(inputStream, options, cancellationToken);
            }
            
            await task;
```

Does it matter (AOT in mind)?

Reply:
The same for the other places below too.